doc/designs/ldap_pam_passthrough.md

```diff
@@ -0,0 +1,315 @@
+   389-ds treats this as no authentication decision so allows other
+   plugins to try.
+
+ - If the user DN is in the allowed configuration for PAM pass-through
+   then the PAM stack is loaded and processed. One can limit the
+   subtrees that may be authenticated. In the IPA case this is basically
+   moot because of the flat DIT.
+
+ - PAM is invoked an dpam_sss.so and attempts to perform a
+   password-based authentication for 'user' account using the provided
+   credentials.
+
+ - Since the user account matches the IPA domain, it will be Kerberos
+   authentication against IPA KDC running on the same host, as we are
+   authenticating on a IPA server.
+
+ - The IPA KDB driver in the KDC will notice that the 'user@IPA' principal
+   has the 'radius' pre-authentication method configured with TL data
+   "otp\0[{\"indicators\": [\"radius\"]}]" meaning it should advertise
+   the OTP pre-auth mechanism to the Kerberos client.
+
+ - The client (SSSD) notices the availability of the OTP pre-auth
+   mechanism uses host principal's TGT as its FAST channel wrapper to
+   proceed with OTP pre-auth.
+
+ - OTP pre-auth mechanism will talk between KDC and the client to ask
+   for additional details (OTP value) via prompting mechanism it has.
+
+ - The client (SSSD) will return the OTP value and the KDC then will
+   issue a RADIUS request "Accept-Request" to a RADIUS server configured
+   in the KDC configuration. The OTP value in this case is the
+   credentials the user provided.
+
+ - ipa-otpd handles OTP requests and will connect to LDAP (the LDAP URI
+   is passed as part of ipa-otpd@.service definition from
+   /etc/ipa/default.conf, so it'll be an LDAPI access). ipa-otpd will
+   parse the RADIUS packet and look up requested user principal entry from
+   LDAP.
+
+ - If the user principal entry has the 'radius' authentication indicator
+   configured (or it is default for IPA deployment) and there is a
+   RADIUS proxy link in the user entry (there is no default so it must
+   be set per user), it will send the same RADIUS packet to the RADIUS
+   server configured as a proxy link with the credentials provided by
+   KDC
+
+ - For a native OTP setup where the user has the 'otp' authentication
+   indicator the process is about the same: instead of sending a RADIUS
+   proxy request, ipa-otpd will bind to LDAP with the user DN and pass
+   the credentials provided by KDC.
+
+## Workaround
+
+Due to the way passwords and OTP are handled by the current IPA
+password plugin it is possible to make this work today:
+
+ - Install IPA
+
+ - Configure the 389-ds PAM pass-through plugin to a PAM service that
+   relies on pam_sss.so (e.g. system-auth)
+
+ - Add a RADIUS proxy configuration in IPA
+
+ - Add this proxy to one or more users
+
+ - set default authentication indicator in IPA to 'radius, otp' or on
+   one or more of those users.
+
+ - Do *not* set a password or principal key for those users
+
+ - The user has no OTP tokens
```

Comment:
Previous items on the list use imperatives, may be turn this one into a similar sentence?
> Do *not* define any OTP tokens for those users

Reply:
done